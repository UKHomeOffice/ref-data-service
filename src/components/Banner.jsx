import React from 'react'

import config from '../../config/core';

const { environment, serviceDesk } = config;

const Banner = (props) =>  {
  const environmentBanner = () => {
    if (environment.toLowerCase() === 'production') {
      return <React.Fragment></React.Fragment>;
    } else {
      return <strong className='govuk-tag govuk-phase-banner__content__tag '>{environment}</strong>;
    }
  }

  return (
    <div className='govuk-phase-banner'>
      <p className='govuk-phase-banner__content'>
        <strong className='govuk-tag govuk-phase-banner__content__tag '>beta</strong>
        {environmentBanner()}
        <span className='govuk-phase-banner__text'>
          This is a new service â€“ your <a className='govuk-link' href={`${serviceDesk.feedback}`}>feedback</a> will help us to improve it.
        </span>
      </p>
    </div>
  );
}

export default Banner;
