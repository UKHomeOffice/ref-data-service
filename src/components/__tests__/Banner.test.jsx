import React from 'react';
import { shallow } from 'enzyme';

// local imports
import Banner from '../Banner';

// mock the configuration object
jest.mock('../../../config/core', () => (
  {
    environment: 'development',
    serviceDesk: {
      feedback: 'https://servicedesk.com/customer/portal/3/create/50'
    },
  }
));

describe('Banner', () => {
  it('Should render banner successfully', () => {
    const wrapper = shallow(<Banner/>);

    expect(wrapper).toMatchSnapshot();
  });
});
