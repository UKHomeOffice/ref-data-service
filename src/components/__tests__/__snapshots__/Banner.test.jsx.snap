// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Should render banner successfully 1`] = `
<div
  className="govuk-phase-banner"
>
  <p
    className="govuk-phase-banner__content"
  >
    <strong
      className="govuk-tag govuk-phase-banner__content__tag "
    >
      beta
    </strong>
    <strong
      className="govuk-tag govuk-phase-banner__content__tag "
    >
      development
    </strong>
    <span
      className="govuk-phase-banner__text"
    >
      This is a new service â€“ your 
      <a
        className="govuk-link"
        href="https://servicedesk.com/customer/portal/3/create/50"
      >
        feedback
      </a>
       will help us to improve it.
    </span>
  </p>
</div>
`;
